src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1890,37 +1882,26 @@ private CorInfoType asCorInfoType(CORINFO_CLASS_STRUCT_* cls)
         }
 
 
-        private int appendClassName(char** ppBuf, ref int pnBufLen, CORINFO_CLASS_STRUCT_* cls, bool fNamespace, bool fFullInst, bool fAssembly)
+        private nuint printClassName(CORINFO_CLASS_STRUCT_* cls, byte* buffer, nuint bufferSize, nuint* pRequiredBufferSize)
         {
-            // We support enough of this to make SIMD work, but not much else.
-
-            Debug.Assert(fNamespace && !fFullInst && !fAssembly);
-
             var type = HandleToObject(cls);
-            string name = TypeString.Instance.FormatName(type);
+            string name = JitTypeNameFormatter.Instance.FormatName(type);
+            byte[] utf8 = Encoding.UTF8.GetBytes(name);
```

Comment:
This can use `Utf8.FromUtf16` + `Encoding.UTF8.GetByteCount` to avoid the allocation (it is the pattern used by `printObjectDescription`).

Reply:
Will factor these.

Does it make sense to make `getFieldName` and `getMethodName` consistent with the new style? It will require some more EE calls to replace the `scopeName` that they are returning additionally today. I suppose it shouldn't matter much since we should not be using them in release anyway.

Reply:
I agree that the `scopeName` argument does not look pretty. Are you thinking about deleting the `scopeName` argument and switching the callers to use the `printClassName` instead?

Reply:
Yes, the callers would switch to `getFieldClass`/`getMethodClass` + `printClassName`  instead.